django/db/migrations/loader.py

```diff
@@ -372,6 +369,20 @@ def check_consistent_history(self, connection):
                         )
                     )
 
+    def all_replaced_applied(self, migration_key, applied):
+        """
+        Checks (recursively) whether all replaced migrations are applied.
```

Comment:
Just noting for others: the checking for cycles is done earlier, in `replace_migration()`. I can't find a way to cause recursion errors that wouldn't be caught there.